.azuredevops/templates/steps/dependencies-rocm.yml

```diff
@@ -141,47 +141,84 @@ parameters:
+  type: string
+  default: ''
+# array of ROCm components that can specify GPU target or have dependency of such a component
+# these components would have parallel build jobs per GPU target so they produce multiple artifacts
+# only download artifact tied to the GPU target
+- name: componentsWithGPUTarget
+  type: object
+  default:
+    - AMDMIGraphX
+    - composable_kernel
+    - hipBLASLt
+    - hipCUB
+    - hipFFT
+    - hipRAND
+    - hipSPARSELt
+    - hipTensor
+    - rccl
+    - rocALUTION
+    - rocBLAS
+    - rocFFT
+    - rocm-examples
+    - rocPRIM
+    - rocprofiler
+    - rocRAND
+    - rocSOLVER
+    - rocSPARSE
+    - rocThrust
+    - roctracer
+    - rocWMMA
+    - rpp
+# list below do not have flag for gpu target but have dependencies of components that do, so build separately per gpu target
+    - hipBLAS
+    - hipSOLVER
+    - hipSPARSE
+    - MIOpen
+    - MIVision
+    - rocAL
+    - ROCmValidationSuite
 
 steps:
 # assuming artifact-download.yml template file in same directory
 # for the case where rocm dependency item in list has a colon (:)
 # assume it is of the format of componentName:fileFilter
 # fileFilter could contain both a subcomponent name or gpu name separated by asterisks
+# gpu name will be specified by parameters.gpuTarget for components that are in componentsWithGPUTarget
 # e.g., gfx942 to only download artifacts from component for this gpu if applicable
 - ${{ each dependency in parameters.dependencyList }}:
   - ${{ if contains(dependency, ':') }}:
-    - ${{ if eq(parameters.dependencySource, 'staging') }}:
-      - template: artifact-download.yml
-        parameters:
-          componentName: ${{ split(dependency, ':')[0] }}
-          pipelineId: ${{ parameters.stagingPipelineIdentifiers[split(dependency, ':')[0]] }}
-          fileFilter: ${{ split(dependency, ':')[1] }}
+    - template: artifact-download.yml
+      parameters:
+        componentName: ${{ split(dependency, ':')[0] }}
+        extractToMnt: ${{ parameters.extractToMnt }}
+        ${{ if eq(parameters.dependencySource, 'staging') }}:
+          pipelineId: ${{ parameters.stagingPipelineIdentifiers[ split(dependency, ':')[0] ] }}
           latestFromBranch: ${{ parameters.latestFromBranch }}
-          extractToMnt: ${{ parameters.extractToMnt }}
-    - ${{ if eq(parameters.dependencySource, 'tag-builds') }}:
-      - template: artifact-download.yml
-        parameters:
-          componentName: ${{ split(dependency, ':')[0] }}
-          pipelineId: ${{ parameters.taggedPipelineIdentifiers[split(dependency, ':')[0]] }}
-          fileFilter: ${{ split(dependency, ':')[1] }}
+        ${{ elseif eq(parameters.dependencySource, 'tag-builds') }}:
+          pipelineId: ${{ parameters.taggedPipelineIdentifiers[ split(dependency, ':')[0] ] }}
           latestFromBranch: false
-          extractToMnt: ${{ parameters.extractToMnt }}
+        ${{ if containsValue( parameters.componentsWithGPUTarget, split(dependency, ':')[0] ) }}:
```

Comment:
Is the `componentsWithGPUTarget` list needed? If a value is passed in for `gpuTarget`, that's already a good indication that it needs a specific gpu target. 

If no value is passed, then `gpuTarget` is an empty string and `fileFilter` isn't affected by it (if `fileFilter` actually does care about the empty string then you can disregard this comment).

Reply:
Not every ROCm component in the dependency array knows what to do with gpuTarget if a value is set though. For example, llvm-project. It will go through incorrect logic if we add a filter by gpuTarget name for artifacts.